ion/src/simulator/shared/main.cpp

```diff
@@ -77,7 +77,13 @@ void init() {
 
   SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "1");
 
+  // Try creating a hardware-accelerated renderer.
   sRenderer = SDL_CreateRenderer(sWindow, -1, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);
+  if (!sRenderer) {
+    // Try creating a software renderer.
+    sRenderer = SDL_CreateRenderer(sWindow, -1, 0);
+  }
+  assert(sRenderer);
 
```

Comment:
I'm kind of wary about this because this code is ran on all platforms.
By the way, shouldn't we simply remove the `SDL_RENDER_ACCELERATED` flag? See https://wiki.libsdl.org/SDL_CreateRenderer#flags

Reply:
It fails too if `SDL_RENDERER_PRESENTVSYNC` is set. I would not be worried about the fallback since, well, it's a fallback and it will be very obvious if it is engaged: scaling is done with a nearest-neighbor algorithm. Better to have an ugly but working output than an assertion failure.